encord/metadata_schema.py

```diff
@@ -268,22 +268,25 @@ def add_scalar(
         self,
         k: str,
         *,
-        data_type: Literal["boolean", "datetime", "number", "uuid", "text", "varchar", "string", "long_string"],
+        data_type: Union[
+            Literal["boolean", "datetime", "number", "uuid", "text", "varchar", "string", "long_string"], str
```

Comment:
IMO this would be collapsed into `data_type: str`, as all the literal values are `str` under the hood

Reply:
the literals i think are there for these are the actual valid values, the str while the actual collapsed type makes the docs less useful.

Reply:
agreed with James, we added an Enum to support type checking on general cases where the input can't be easily categorised as Literal[...]. This way users don't have to silence mypy anymore and know what the valid inputs are.